lib/cli/deploy.rs

```diff
@@ -11,24 +11,45 @@ pub fn with_payment_and_session(
     deploy_params: DeployStrParams,
     payment_params: PaymentStrParams,
     session_params: SessionStrParams,
+    allow_unsigned_deploy: bool,
 ) -> Result<Deploy, CliError> {
     let chain_name = deploy_params.chain_name.to_string();
     let session = parse::session_executable_deploy_item(session_params)?;
-    let secret_key = parse::secret_key_from_file(deploy_params.secret_key)?;
+    let maybe_secret_key = if allow_unsigned_deploy && deploy_params.secret_key.is_empty() {
+        None
+    } else if deploy_params.secret_key.is_empty() && !allow_unsigned_deploy {
+        return Err(CliError::InvalidArgument {
+            context: "with_payment_and_session (secret_key, allow_unsigned_deploy)",
+            error: format!(
+                "allow_unsigned_deploy was {}, but no secret key was provided",
+                allow_unsigned_deploy
+            ),
+        });
+    } else {
+        Some(parse::secret_key_from_file(deploy_params.secret_key)?)
+    };
```

Comment:
```suggestion
    let maybe_secret_key = if !deploy_params.secret_key.is_empty() {
        Some(parse::secret_key_from_file(deploy_params.secret_key)?)
    } else {
        if !allow_unsigned_deploy {
            return Err(CliError::InvalidArgument {
                context: "with_payment_and_session (secret_key, allow_unsigned_deploy)",
                error: format!(
                    "allow_unsigned_deploy was {}, but no secret key was provided",
                    allow_unsigned_deploy
                ),
            });
        }
        None
    };
```

the if else is not very clear to read so I suggest you isolate your case first then decline the rest